### PR TITLE
add dms v2 kafka instances in sdk

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
add dms v2 kafka instances in sdk

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: openstack/dms/v2/kafka/instances/requests.go [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+	// consists of 4 to 64 characters, and supports
+	// only letters, digits, hyphens (-), and underscores (_).
+	Name string `json:"name" required:"true"`
+
+	// Indicates the description of an instance.
+	// It is a character string containing not more than 1024 characters.
+	Description string `json:"description,omitempty"`
+
+	// Indicates a message engine.
+	// Currently, only RabbitMQ is supported.
```
Comment (ShiChangkuo) [inline openstack/dms/v2/kafka/instances/requests.go:27]: this comment may be misleading

File: openstack/dms/v2/kafka/instances/requests.go [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+	ConnectorEnalbe bool `json:"connector_enable,omitempty"`
+
+	// Indicates whether to enable automatic topic creation.
+	EnableAutoTopic bool `json:"enable_auto_topic,omitempty"`
+
+	//Indicates the storage I/O specification. For details on how to select a disk type
+	StorageSpecCode string `json:"storage_spec_code,omitempty"`
+
+	// Indicates the enterprise project ID.
+	EnterpriseProjectID string `json:"enterprise_project_id,omitempty"`
```
Comment (ShiChangkuo) [inline openstack/dms/v2/kafka/instances/requests.go:124]: **tags** is missing and suggest to import `ResourceTags` in openstack/common/tags package

File: openstack/dms/v2/kafka/instances/requests.go [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+type UpdateOpts struct {
+	// Indicates the name of an instance.
+	// An instance name starts with a letter,
+	// consists of 4 to 64 characters,
+	// and supports only letters, digits, and hyphens (-).
+	Name string `json:"name,omitempty"`
+
+	// Indicates the description of an instance.
+	// It is a character string containing not more than 1024 characters.
+	Description string `json:"description,omitempty"`
```
Comment (ShiChangkuo) [inline openstack/dms/v2/kafka/instances/requests.go:170]: change the type to ***string** then we can clear the description field.

File: openstack/dms/v2/kafka/instances/requests.go (hunk 1 above)
Comment (Jason-Zhang9309) [reply openstack/dms/v2/kafka/instances/requests.go:27]: fixed

File: openstack/dms/v2/kafka/instances/requests.go (hunk 3 above)
Comment (Jason-Zhang9309) [reply openstack/dms/v2/kafka/instances/requests.go:170]: fixed

File: openstack/dms/v2/kafka/instances/requests.go (hunk 2 above)
Comment (Jason-Zhang9309) [reply openstack/dms/v2/kafka/instances/requests.go:124]: fixed

**Pull request merged**